tutorials/eth-api/call-permit-gasless-txs.md

```diff
@@ -0,0 +1,442 @@
+
+The domain separator and the typed data structure will be based on the [Call Permit Precompile](/builders/pallets-precompiles/precompiles/call-permit){target=_blank}. The steps to build both of these components will always be the same, regardless of the data that is being signed. The actual message data is what will change depending on your individual use case.
+
+### Define the Domain Separator {: #define-domain-separator }
+
+We'll first start off with the domain separator, which will define the Call Permit Precompile as the signing domain. Permits will get dispatched by calling the `dispatch` function of the Call Permit Precompile, which is why the Call Permit Precompile is always going to be the signing domain. As previously mentioned, the goal of the domain separator is to avoid replay attacks. 
+
+--8<-- 'text/precompiles/call-permit/domain-separator.md'
+
+We're using Ethers in this example, which requires the domain separator to be in the format specified by the [`TypedDataDomain` interface](https://docs.ethers.org/v6/api/hashing/#TypedDataDomain){target=_blank}, but if desired, you could generate the domain separator as a *bytes32* representation using the [`DOMAIN_SEPARATOR()` function of the Call Permit Precompile](/builders/pallets-precompiles/precompiles/call-permit/#:~:text=DOMAIN_SEPARATOR()){target=_blank}.
+
+The domain separator for each Moonbeam network is as follows:
+
+=== "Moonbeam"
+
+    ```js
+    const domain = {
+      name: 'Call Permit Precompile',
+      version: '1',
+      chainId: {{ networks.moonbeam.chain_id }},
+      verifyingContract: '{{ networks.moonbeam.precompiles.call_permit}}',
+    };
+    ```
+
+=== "Moonriver"
+
+    ```js
+    const domain = {
+      name: 'Call Permit Precompile',
+      version: '1',
+      chainId: {{ networks.moonriver.chain_id }},
+      verifyingContract: '{{ networks.moonriver.precompiles.call_permit}}',
+    };
+    ```
+
+=== "Moonbase Alpha"
+
+    ```js
+    const domain = {
+      name: 'Call Permit Precompile',
+      version: '1',
+      chainId: {{ networks.moonbase.chain_id }},
+      verifyingContract: '{{ networks.moonbase.precompiles.call_permit}}',
+    };
+    ```
+
+=== "Moonbeam Dev Node"
+
+    ```js
+    const domain = {
+      name: 'Call Permit Precompile',
+      version: '1',
+      chainId: {{ networks.development.chain_id }},
+      verifyingContract: '{{ networks.moonbase.precompiles.call_permit}}',
+    };
+    ```
+
+### Define the Typed Data Structure {: #define-typed-data-structure }
+
+Next, we'll need to define the typed data structure. The typed data structure defines the acceptable types of data that our users will be signing. If you take a look at the [`dispatch` function of the Call Permit Precompile](/builders/pallets-precompiles/precompiles/call-permit/#the-call-permit-interface){target=_blank}, you'll see the data that we need to send, along with the associated types, is as follows:
+
+```sol
```

Comment:
Maybe add a bullet list of what each input means? not sure what do you think?

Update: I noticed you explain them after, but I would add a note saying that you do :)

Reply:
yeah this section is primarily about the data types and the following actually goes into the actual data and values being passed in. I'll add a note!